Source/JavaScriptCore/runtime/JSString.h

```diff
@@ -985,70 +987,115 @@ ALWAYS_INLINE JSString* jsString(VM& vm, Ref<StringImpl>&& s)
     return jsString(vm, String { WTFMove(s) });
 }
 
-inline JSString* jsSubstring(VM& vm, JSGlobalObject* globalObject, JSString* base, unsigned offset, unsigned length)
+inline JSString* JSString::substringImpl(VM& vm, JSGlobalObject* globalObject, unsigned offset, unsigned length)
```

Comment:
How about naming it to trySubsubstringImpl to make it explicit that it can fail.